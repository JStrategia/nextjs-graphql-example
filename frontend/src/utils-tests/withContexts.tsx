import React from 'react';
import ThemeProvider from 'contexts/ThemeProvider';

function withContexts(children: React.ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
