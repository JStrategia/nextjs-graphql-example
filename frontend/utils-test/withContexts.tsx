import React, { ReactNode } from 'react';
import ThemeProvider from '../contexts/ThemeProvider';

function withContexts(children: ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
