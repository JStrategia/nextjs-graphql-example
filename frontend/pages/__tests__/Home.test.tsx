import * as React from 'react';
import { create } from 'react-test-renderer';
import Home from '../index.page';
import withContexts from '../../utils-test/withContexts';

describe('Home', () => {
  describe('When rendering', () => {
    const render = create(withContexts(<Home />)).toJSON();

    test('Then match snapshot', () => {
      expect(render).toMatchSnapshot();
    });
  });
});
