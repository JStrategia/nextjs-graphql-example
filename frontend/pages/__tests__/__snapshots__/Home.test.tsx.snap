// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home When rendering Then match snapshot 1`] = `
<div
  className="sc-bczRLJ eGDUch"
>
  <div
    className="sc-gsnTZi ksunwW"
  >
    <main>
      <h1
        className="sc-dkzDqf gkDXsv"
      >
        nextRecipes
      </h1>
      <span
        className="sc-hKMtZM gnmBBv"
      >
        websiteAboutRecipes
      </span>
    </main>
  </div>
</div>
`;
